### DISCUSSION TITLE
Discuss its performance characteristics

### DISCUSSION THREAD

**Discussion by makoConstruct** [risc0/risc0#96] [General]
- Whether the performance of risc0 is linearly proportional to a regular risc processor
  - If so, what the proportion is, or what we can expect it to reach after sufficient optimization.
  - If not, what characteristics of the risc programs will determine their runtime in risc0? I couldn't guess.

Without knowing these things, it's difficult to estimate the impact or usefulness of risc0.

**Comment by zicklag** [comment]
I know that risc0 is still under heavy development, and you are actually doing very well as far as documentation goes for a project that is so new, but I would also be curious about some performance statistics, such as the size and performance of the proofs for the examples.

Especially with this line in the docs:

> The way RISC Zero achieves this is by inventing a uniquely high-performance ZKP prover and then using the performance headroom to build a zero knowledge virtual machine (ZKVM) implementing a standard RISC-V instruction set.

I'd be curious how risc0's performance compares to other STARK implementations.

But again, you guys are doing great with the website and docs, keep up the good work and if you need to develop more first, by all means do so!

**Comment by jbruestle** [comment]
Technically one of the proof steps is N log(N), but the majority of the time is actually spent in the O(N) parts, so generally proof speed is roughly linear with with number of cycles that the RISC-V executes, and on my MacBook Pro that's about 30K cycles of the RISC-V per second.
